docs/okeps/okep-5674-dpu-healthcheck.md

```diff
@@ -0,0 +1,159 @@
+
+```mermaid
+flowchart TB
+
+  subgraph HOST[Host]
+    direction TB
+    H1["Kubelet"]
+    H2["ovn-kubernetes (DPU-Host mode)"]
+  end
+
+  subgraph DPU[DPU]
+    direction TB
+    D1["ovn-kubernetes (DPU mode)"]
+    D2[OVS]
+    D3[OVN]
+    D1 --> D2
+    D1 --> D3
+  end
+
+  HOST <--> DPU
+```
+
+Note, "trusted" mode here means the DPU side has the kubeconfig of the host cluster, and runs OVNK in the DPU. There
+is an "untrusted" mode, but it is outside the scope of this enhancement and will not use this feature.
+
+As can be seen in the diagram, the ovn-kubernetes DPU mode is running down in the DPU, managed outside the purview of
+the host Kubernetes cluster. Therefore, the Host Kubernetes cluster has no observability or failure detection of this pod.
+When a DPU accelerated pod is created, the following happens:
+
+1. Kubelet invokes Multus which checks that the UDN/NAD has a resourceName field
+2. SR-IOV plugin is called and reserves a VF, passes the PCI adddress in the CNI ADD
+3. OVN-Kubernetes DPU-Host receives the CNI ADD, and annotates the pod with the connection details.
+4. OVN-Kubernetes DPU waits for the annotation, plugs the VF representor into OVS, and annotates the pod with typical
+k8s.ovn.org annotations.
+5. OVN-Kubernetes DPU-Host sees annotations, and then moves the VF into the pod and configures it
+6. DPU-Host sends CNI ADD reply
+
+Now, if the DPU side goes down, DPU-Host will keep waiting on the CNI ADD for 2 minutes. All CNI ADDs will keep failing.]
+From a user's perspective on the Host Kubernetes cluster, they may have no insight into what is running in the DPU from
+KAPI. We can make this better by:
+
+1. Detecting when the DPU side has gone down.
+2. Failing CNI operations fast as we know when the DPU is down.
+3. Set the node as NetworkUnavailable so that Kubelet will stop scheduling pods here.
+
+## User-Stories/Use-Cases
+
+Story 1: Make DPU operations more robust
+
+As a cluster admin, I want to leverage DPUs for hardware offload. However, I do not want to compromise my
+visibility into the cluster and be stuck with components that are failing outside my view. I'd like to have DPU
+issues propagate up the stack so that I can see when a node has failed. Additionally, I would like Kubelet to no longer
+schedule pods to this node until it has come back online.
+
+## Proposed Solution
+
+Kubernetes uses node leases as a mechanism to maintain health within a cluster. This solution proposes that a custom
+node lease is created and used by OVN-Kubernetes to detect when a DPU is no longer functioning. The DPU OVN-Kubernetes
+component will update this node lease periodically. DPU-Host will monitor this node lease, and when it expires it will:
+
+1. Trigger CNI operations to fail fast, CNI is down.
+2. Update the node condition with NetworkUnavailable.
+
+Kubelet will see the NetworkUnavailable and stop scheduling pods to this node. This will not prevent OVN-Kubernetes pods
```

Comment:
I'm going to chat to someone and confirm this flow, but I'm not 100% sure this s accurate. We don't want kubelet to stop scheduling pods to the node - because hostnetworked pods are fair game.